documentation/cadl-structure-guidelines.md

```diff
@@ -0,0 +1,157 @@
+
+## Structure Overview
+
+This proposal strives to align with [Azure SDK Repo Structure Guidelines](https://azure.github.io/azure-sdk/policies_repostructure.html).
+
+We eliminate the distinction between data-plane and resource-manager, treating all packages as siblings under the service family folder. This is what the SDKs do today. The legacy `data-plane` and `resource-manager` folders where Swagger specs live will remain.
+
+A given Cadl folder could represent any of the following scenarios:
+
+- an SDK and service spec (simplest, common for small services)
+- a service spec only
+- an SDK spec that aggregates many small service specs (one SDK to many microservices)
+- an SDK spec that extracts a portion of a large service spec (many SDKs to a single large service)
+- a shared library that represents neither a service spec nor an SDK
+
+Examples below:
+
+```
+-> specification
+   -> confidentialledger
+      -> ConfidentialLedger               (data-plane SDK + service)
+      -> data-plane                       (swagger)
+      -> resource-manager                 (swagger)
+   -> compute
+      -> Compute.Management               (mgmt SDK + service)
+      -> Compute.Management.Shared        (shared)
+      -> Compute.CloudService.Management  (service)
+      -> Compute.Diagnostic.Management    (service)
+      -> Compute.Disk.Management          (service)
+      -> Compute.Gallery.Management       (service)
+      -> Compute.Skus.Management          (service)
+      -> data-plane                       (swagger)
+      -> resource-manager                 (swagger)
+   -> keyvault
+      -> KeyVault.Certificates            (data-plane SDK + service)
+      -> KeyVault.Keys                    (data-plane SDK + service)
+      -> KeyVault.Secrets                 (data-plane SDK + service)
+      -> KeyVault.Management              (mgmt SDK + service)
+      -> data-plane                       (swagger)
+      -> resource-manager                 (swagger)
+```
+
+## Service Folders
+
+The service folder contains the entire Cadl library specification for a service, which could include custom linter rules, methods, etc.
+
+The folder name should correspond to the RP-name, but dropping the leading "Azure" or "Microsoft" prefix. Two additional naming conventions should be following:
+
+- `Management` at the end of the service RP-name indicates a management (resource manager) libarary.
+- `Shared` at the end of the name indicates a shared library. If a management library has a shared component (unlikely), `Shared` should follow `Management`.
+
+### Packages
+
+All services and service family libraries are modeled as Cadl packages, since you must install supporting librarie for proper tooling support. Per this proposal, all packages defined in the repo would be **unpublished**. Only packages in `cadl-azure` would be published. We will annotate the packages with `"private": true` in the `package.json` file to prevent publishing these to npm.
```

Comment:
Can we mention that how service team is supposed to consume these **unpublished** packages? What is the version control and support level for these packages when they're referenced by other services?

Reply:
This is covered in the proposal. Specs should not need to be consumed, but shared libraries are consumed using relative file reference syntax.

Reply:
In cross service reference scenario, will shared libraries version upgrade break existed reference scenario? Will this be restricted by breaking change rule?